src/staker.ts

```diff
@@ -0,0 +1,149 @@
+   */
+  tokenId: BigintIsh
+
+  /**
+   * Address to send rewards to.
+   */
+  recipient: string
+
+  /**
+   * The amount of `rewardToken` to claim. 0 claims all.
+   */
+  amount?: BigintIsh
+
+  /**
+   * Set when withdrawing. The position will be sent to `owner` on withdraw.
+   */
+  owner?: string
+
+  /**
+   * Set when withdrawing. `data` is passed to `safeTransferFrom` when transferring the position from contract back to owner.
+   */
+  data?: string
+}
+
+export abstract class Staker {
+  public static INTERFACE: Interface = new Interface(abi)
+
+  protected constructor() {}
+
+  /**
+   *  To claim rewards, must unstake and then claim.
+   * @param incentiveKey The unique identifier of a staking program.
+   * @param options Options for producing the calldata to claim. Can't claim unlesss you unstake.
+   * @returns The calldatas for 'unstakeToken' and 'claimReward'.
+   */
+  private static encodeClaim(incentiveKey: IncentiveKey, options: ClaimOptions): string[] {
+    const calldatas: string[] = []
+    calldatas.push(
+      Staker.INTERFACE.encodeFunctionData('unstakeToken', [
+        {
+          key: incentiveKey,
+          tokenId: options.tokenId
+        }
+      ])
+    )
+    const recipient: string = validateAndParseAddress(options.recipient)
+    calldatas.push(
+      Staker.INTERFACE.encodeFunctionData('claimReward', [
+        {
+          rewardToken: incentiveKey.rewardToken,
+          to: recipient,
+          amountRequested: options.amount
+        }
+      ])
+    )
+    return calldatas
+  }
+
+  /**
+   * Only claims rewards. Must unstake, claim, and then restake.
+   */
+  public static collectRewards(incentiveKey: IncentiveKey, options: ClaimOptions): MethodParameters {
+    const calldatas = this.encodeClaim(incentiveKey, options)
+    calldatas.push(
+      Staker.INTERFACE.encodeFunctionData('stakeToken', [
+        {
+          key: incentiveKey,
+          tokenId: options.tokenId
+        }
+      ])
+    )
+    return {
+      calldata: Staker.INTERFACE.encodeFunctionData('multicall', [calldatas]),
+      value: '0x'
+    }
+  }
+
+  /**
+   * Creates claim and unstake calldata. Option to withdraw token.
+   * Note: Only really makes sense to either claim and continue staking or claim and withdraw.
+   * Note:
+   */
+  /**
+   *
+   * @param incentiveKeys A list of incentiveKeys to unstake from. Should include all incentiveKeys (unique staking programs) that `options.tokenId` is staked in.
+   * @param options Options for producing collect/claim calldata. Can't withdraw without unstaking all programs for `tokenId`.
+   * @returns Calldata for unstaking, claiming, and withdrawing.
+   */
+  public static withdrawToken(incentiveKeys: IncentiveKey[], options: ClaimOptions): MethodParameters {
+    const calldatas: string[] = []
+    for (let i = 0; i < incentiveKeys.length; i++) {
+      const incentiveKey = incentiveKeys[i]
+      calldatas.concat(this.encodeClaim(incentiveKey, options))
```

Comment:
prefer `push` over inline `concat` i think

Reply:
ok! I only did concat because `encodeClaim` returns string[] which is the wrong input type for push. In this case would just take a sting. So I could do concat() or I could do push twice:
`const calldata = this.encodeClaim(incentiveKey, claimOptions)
 calldatas.push(calldata[0])
 calldatas.push(calldata[1])
`
What do you think?

Reply:
oh i see, never mind then, concat is fine